pkg/clients/azure_test.go

```diff
@@ -28,6 +28,7 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/google/go-cmp/cmp"
 	"github.com/onsi/gomega"
+	"github.com/stretchr/testify/assert"
```

Comment:
Could I convince you to use `cmp.Diff` rather than `assert.Equal`?

We typically subscribe to the upstream Go position of preferring plain old `testing.T` (plus a comparison/diff helper like `cmp`) over third party assertion and testing libraries. The idea being that when you're using a third party library contributors need to learn the DSL of that library, whereas most folks are familiar with the stdlib `testing` package. This problem is exacerbated if there are multiple testing libraries used within the one project.

I'm definitely happy for each provider to form opinions and patterns of its own in future once each provider has its own team of maintainers, but for now while my team is doing most of the maintenance I'd prefer to stick to our existing conventions.

https://github.com/golang/go/wiki/TestComments#assert-libraries

Reply:
This assertion library is used in another project I contribute too and seems to be very intuitive IMO. I agree that adding more "non-standard" libraries raises the bar for contribution but this one is minor and does not disrupt the project.

I will stick to the decision that was made since consistency has value. For that reason, I will change the code to stop using the new library.

Reply:
Thanks, I appreciate it!